packages/http/src/validator/validators/body.ts

```diff
@@ -29,6 +29,30 @@ export function deserializeFormBody(
     return decodedUriParams;
   }
 
+  // if the request body array contains JSON objects with multiple fields (i.e. '{"foo":"value"}, {"foo":"dd","xx":"xx"}'),
+  // depending on the deserialization indicated in the encoding, it's possible we end up with an array of broken JSON
+  // objects that were split on the ',' character, such as [ '{"foo":"value"}', '{"foo":"dd"', '"xx":"xx"}' ]. This function
+  // processes such cases so that complete JSON objects, i.e. [ '{"foo":"value"}', '{"foo":"dd", "xx":"xx"}' ], are handled
+  function parseBrokenJSONArray(inputArray: string[]): string[] {
+    let parsedJSONObjects: any[] = [];
+    let currentJSONObject: string = "";
+  
+    for (let item of inputArray) {
+      // handle the scenario where a JSON object in the encoded array is preceded by a "+", which can occur when 
+      // the user puts a space between JSON array entries, such as '{"foo": "a"}, {"foo":"b"}'
+      item = item.startsWith("+") ? item.substring(1) : item;
+      currentJSONObject += (currentJSONObject.length > 0 ? "," : "") + item;
+
+      try {
+        const parsed = JSON.parse(currentJSONObject);
+        parsedJSONObjects.push(parsed);
+        currentJSONObject = "";
+      } catch (_) {}
```

Comment:
This is not actually a noop. This algorithm is attempting to piece together an array consisting of broken pieces of JSON objects. The try/catch paradigm allows us to only process the JSON object when the JSON.parse succeeds, and we otherwise continue accumulating the object in line 41 until we've built the entire object. So we do NOT want to do anything (hence the empty catch statement) until we have successfully built up the entire JSON object in line 41, which will then get successfully added to the parsedJSONObjects list in line 44.

Reply:
However, you make a good point about error reporting for an invalid JSON input string. This scenario actually isn't related to whether parsedJSONObjects is empty, but has to do with whether currentJSONObject is non-empty when we exit this function. If currentJSONObject is NOT empty at the end of parsing, it means there was an error in the input JSON array string that we were not able to parse. I've thus added a 415 error based on this logic starting on line 72 of this file, just below this comment.